programs/svm-spoke/src/utils/merkle_proof_utils.rs

```diff
@@ -1,11 +1,16 @@
 use anchor_lang::{prelude::*, solana_program::keccak};
 
-use crate::{common::V3RelayData, error::CommonError};
+use crate::{common::V3RelayData, error::CommonError, utils::hash_non_empty_message};
 
 pub fn get_v3_relay_hash(relay_data: &V3RelayData, chain_id: u64) -> [u8; 32] {
     let mut input = relay_data.try_to_vec().unwrap();
+
+    // Trim off the serialized message (4 bytes length + message bytes) as it is replaced with the hash.
```

Comment:
why do we need to do this?

Reply:
Alternative is to serialize all struct fields manually, except for message and then serialize the message hash instead. But that is error prone and also turned out to be more CU intensive than this implementation where we serialize the whole struct with original message, trim off the message and replace it with message hash. I can add bit more comments on this, but that would be more than 1 line.